### PR TITLE
#161255198 articles CRUD

### PR DESCRIPTION
**Pull request by p8ul**
### What does this PR do?
- Creates article CRUD

### Description of Task to be completed?
- allow users to list, create, update, delete articles
- test components

### How should this be manually tested?
```yarn run test```

### What are the relevant pivotal tracker stories?
[#161255198](https://www.pivotaltracker.com/story/show/161255198)

##### Article form
screen shot 2018-12-14 at 13 49 39

##### List articles
screen shot 2018-12-14 at 13 52 35

### Checklist:
- [x] My code follows the style guidelines of this project
- [ ] My teammates have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] My PR has one commit.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-REACT_APP_API_URL=http://localhost:8000/api/
\ No newline at end of file
+# REACT_APP_API_URL=https://ah-map-staging.herokuapp.com/api/
+REACT_APP_API_URL=http://127.0.0.1:8000/api/
```
Comment (Rodgers-M) [inline .env]: `.env `  should not be pushed to github. you should add it to `.gitignore`

File: src/components/Article/ArticleImage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ImageUploader from 'react-images-upload';
+
+function ArticleImage(props) {
```
Comment (Rodgers-M) [inline src/components/Article/ArticleImage.js]: rewrite this in es6 please

File: src/components/Article/ArticleTitle.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Form } from 'semantic-ui-react';
+
+function ArticleTitle(props) {
```
Comment (Rodgers-M) [inline src/components/Article/ArticleTitle.js]: this also should be in es6

File: src/components/Article/Editor.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
```
Comment (Rodgers-M) [inline src/components/Article/Editor.js]: what purpose does this serve? i have seen you have another file named `ArticleEditor`

**Review by Rodgers-M — CHANGES_REQUESTED** [review]
see comments. Also all functions should be written using `es6` syntax. 

Note.
you have so many files changed. next time break your story into smaller tasks and raise a PR for each task.

File: .env (hunk 1 above)
Comment (p8ul) [reply .env]: I have removed it, It is already in the .gitignore file.

File: src/components/Article/Editor.js (hunk 4 above)
Comment (p8ul) [reply src/components/Article/Editor.js]: Thank you for the feedback. I have realized that I was repeating myself. I have removed it.

File: src/components/Article/ArticleCard.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      </div>
+      <div className="extra content">
+        <span className="right floated">
+          {readingTime}
+        </span>
+        <span className="float-left">
+          <i className="user icon" />
+          {author.followers.length}
+          {' '}
+Followers
```
Comment (Rodgers-M) [inline src/components/Article/ArticleCard.js:36]: are these followers of the article or the author? am not sure this should be the case unless it was a design decision.

File: src/components/Article/ServerMessage.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Message from '../Signup/Message';
+
+function ServerMessage(props) {
```
Comment (Rodgers-M) [inline src/components/Article/ServerMessage.js]: this function should be in es6

File: src/components/Article/ArticleCard.js (hunk 5 above)
Comment (p8ul) [reply src/components/Article/ArticleCard.js:36]: Its the author follower. This is from the [mockup](http://github.andela.com/ah-maps/) we created during Sims 1. http://github.andela.com/ah-maps/
screen shot 2018-12-16 at 13 23 09

File: src/components/Article/ServerMessage.js (hunk 6 above)
Comment (p8ul) [reply src/components/Article/ServerMessage.js]: Thank you. Rewrote it in es6.

**Review by MartinKaburu — APPROVED** [review]
Looks good to me.

**Review by MartinKaburu — APPROVED** [review]
Looks good to me.

**Review by Tony-Ndichu — APPROVED** [review]
Good job on adding permissions we can all use such as "isOwner"

**Review by mmosoroohh — APPROVED** [review]

**Review by Rodgers-M — APPROVED** [review]

**Pull request merged**